mule-extensions-api-persistence/src/test/java/org/mule/extension/api/persistence/ExtensionModelPersistenceTestCase.java

```diff
@@ -0,0 +1,193 @@
+ * LICENSE.txt file.
+ */
+package org.mule.extension.api.persistence;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+import static org.mule.extension.api.persistence.PersistenceConstants.DISPLAY_MODEL_PROPERTY;
+
+import org.mule.api.connection.ConnectionProvider;
+import org.mule.extension.api.introspection.ConnectionProviderFactory;
+import org.mule.extension.api.introspection.ExpressionSupport;
+import org.mule.extension.api.introspection.ExtensionModel;
+import org.mule.extension.api.introspection.ImmutableConnectionProviderModel;
+import org.mule.extension.api.introspection.ImmutableExtensionModel;
+import org.mule.extension.api.introspection.ImmutableOperationModel;
+import org.mule.extension.api.introspection.ImmutableParameterModel;
+import org.mule.extension.api.introspection.ImmutableRuntimeConnectionProviderModel;
+import org.mule.extension.api.introspection.ImmutableRuntimeExtensionModel;
+import org.mule.extension.api.introspection.ModelProperty;
+import org.mule.extension.api.introspection.property.DisplayModelProperty;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.metadata.java.JavaTypeLoader;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ExtensionModelPersistenceTestCase
+{
+
+    private static final NonExternalizableModelProperty nonExternalizableModelProperty = new NonExternalizableModelProperty();
+    private static final ExternalizableModelProperty externalizableModelProperty = new ExternalizableModelProperty();
+    private static final Set<ModelProperty> modelProperties = new HashSet<>(Arrays.asList(nonExternalizableModelProperty, externalizableModelProperty));
+    private static final JavaTypeLoader javaTypeLoader = new JavaTypeLoader(ExtensionModelPersistenceTestCase.class.getClassLoader());
+    private static final MetadataType stringType = javaTypeLoader.load(String.class);
+    private static final String SERIALIZED_DISPLAY_MODEL_PROPERTY = "{\"displayName\":\"Car Name\",\"password\":false,\"text\":true,\"order\":0}";
+    private static final String GET_CAR_OPERATION_NAME = "getCar";
+    private static final String CAR_NAME_PARAMETER_NAME = "carName";
+    private static final String MODEL_PROPERTIES_NODE = "modelProperties";
+    private static final String OPERATIONS_NODE = "operations";
+    private static final String PARAMETER_MODELS_NODE = "parameterModels";
+
+    private static ExtensionModel deserializedExtensionModel;
+    private static ImmutableExtensionModel originalExtensionModel;
```

Comment:
i think it's the other way around. You have to do it on a BeforeClass because they're static. Why can't this be done with instance variables and a regular before() ?